$MirOS: ports/net/openvpn/patches/patch-init_c,v 1.1 2006/07/27 11:51:43 tg Exp $

First hunk removes srandom() call.

This patch makes the following work more smooth,
because the 'tail -f' only lets mksh read the
second-to-last line (depends on "daemon ovpn")

|	#!/bin/mksh
|	export PATH=/sbin:/bin:/usr/bin:/usr/sbin:/usr/mpkg/sbin
|
|	function waitforit
|	{
|		tail -f /var/log/messages | while read line; do
|			[[ $line = \
|			    *@( ovpn\[)*@(: Initialization Sequence Completed)* ]] \
|			    && break
|		done
|	}
|
|	function runvpn
|	{
|		waitforit &
|		openvpn --config ovc.cfg $* || exit 1
|		wait
|	}
|
|	cd $(dirname "$0")
|	if [[ $1 = @(r|R)* ]]; then
|		runvpn --redirect-gateway def1
|	else
|		runvpn
|	fi
|	# insert post-init stuff here
|	exit 0

--- src/openvpn/init.c.orig	Tue Jun  2 08:00:59 2015
+++ src/openvpn/init.c	Sun Jul 19 13:41:01 2015
@@ -595,8 +595,6 @@ init_static (void)
   crypto_init_dmalloc();
 #endif
 
-  init_random_seed ();		/* init random() function, only used as
-				   source for weak random numbers */
   error_reset ();		/* initialize error.c */
   reset_check_status ();	/* initialize status check code in socket.c */
 
@@ -1263,6 +1261,7 @@ initialization_sequence_completed (struc
     }
   else
     msg (M_INFO, "%s", message);
+  msg (M_INFO, "%s", "done.");
 
   /* Flag connection_list that we initialized */
   if ((flags & (ISC_ERRORS|ISC_SERVER)) == 0 && connection_list_defined (&c->options))
